packages/scm-extra/src/browser/history/scm-history-widget.tsx

```diff
@@ -220,20 +220,30 @@ export class ScmHistoryWidget extends ScmNavigableListWidget<ScmHistoryListNode>
             if (this.historySupport) {
                 try {
                     const currentCommits = this.status.state === 'ready' ? this.status.commits : [];
+                    const historyOptions = { ...options };
 
-                    let sourceHistory = await this.historySupport.getCommitHistory(options);
+                    // if we already have commits, get the latest one plus one more, we need at least one entry, otherwise an error is thrown
+                    // when fetching the log
+                    if (currentCommits.length > 1) {
+                        historyOptions.range = { fromRevision: currentCommits[1].commitDetails.id, toRevision: 'HEAD' };
+                    }
+
+                    let history = await this.historySupport.getCommitHistory(historyOptions);
                     if (token.isCancellationRequested || !this.hasMoreCommits) {
                         return;
                     }
 
-                    if (options && ((options.maxCount && sourceHistory.length < options.maxCount) || (!options.maxCount && currentCommits))) {
+                    if (options && ((options.maxCount && history.length < options.maxCount) || (!options.maxCount && currentCommits))) {
                         this.hasMoreCommits = false;
                     }
+
+                    // remove the extra commit
                     if (currentCommits.length > 0) {
-                        sourceHistory = sourceHistory.slice(1);
+                        history = history.slice(0, -1);
```

Comment:
Why not?

```suggestion
                        history.pop();
```

So `history` could also be a const again.

Reply:
ah good catch